graph/graph.go

```diff
@@ -15,6 +15,13 @@ type Node interface {
 type Edge interface {
 	From() Node
 	To() Node
+}
+
+// WeightedEdge is a weighted graph edge. In directed graphs, the direction
+// of the edge is given from -> to, otherwise the edge is semantically
+// unordered.
+type WeightedEdge interface {
+	Edge
 	Weight() float64
```

Comment:
That is fine, I agree it is rather redundant. The main reason I prefer to have methods of commonly used interfaces documented, is so that user-defined implementations of those method sets, may include the comment easily and then focus on documenting and extending on what differs from the basic form of the method, if anything.